tests/graphql/client.spec.ts

```diff
@@ -1,84 +1,563 @@
+      .stub(subscriptions, "SubscriptionClient")
+      .returns(fakeSubscriptionClient);
+  });
+
+  after(() => {
+    subscriptionClientStub.restore();
+  });
+
+  describe("when auth token is missing", () => {
+    it("should throw a UserUnauthorized error", () => {
+      let error;
+      try {
+        client.graphQL.createSubscriptionClient();
+      } catch (err) {
+        error = err;
+      }
+
+      expect(error).to.be.instanceof(UserUnauthorizedError);
+    });
+  });
+
+  describe("when auth token is present", () => {
+    before(() => {
+      client.auth.tokenManager.setToken("dummy-token");
+    });
+
+    it("should create a new SubscriptionClient and return it", () => {
+      const subscriptionClient = client.graphQL.createSubscriptionClient();
+      expect(subscriptionClientStub.callCount).to.equal(1);
+      const [endpoint, options, websocket] = subscriptionClientStub.getCall(
+        0
+      ).args;
+
+      expect(endpoint).to.equal(
+        `${KONTIST_SUBSCRIPTION_API_BASE_URL}/api/graphql`
+      );
+      expect(options.lazy).to.equal(true);
+      expect(options.connectionParams).to.equal(
+        client.graphQL.getConnectionParams
+      );
+      expect(websocket).to.equal(ws);
+      expect(subscriptionClient).to.equal(fakeSubscriptionClient);
+    });
+
+    describe("when executing in a browser environment", () => {
+      before(() => {
+        (global as any).window = {
+          WebSocket: { fake: "websocket" }
+        };
+      });
+
+      after(() => {
+        (global as any).window = undefined;
+      });
 
-      const { error, stub } = await setup(response);
+      it("should use the native browser WebSocket implementation", () => {
+        subscriptionClientStub.resetHistory();
 
-      expect(error).to.be.an.instanceof(GraphQLError);
-      expect(error.message).to.equal(message);
-      expect(error.status).to.be.undefined;
-      expect(error.type).to.be.undefined;
+        client.graphQL.createSubscriptionClient();
 
-      stub.restore();
+        expect(subscriptionClientStub.callCount).to.equal(1);
+        expect(subscriptionClientStub.getCall(0).args[2].fake).to.equal(
+          "websocket"
+        );
+      });
     });
   });
 });
+
+describe("getConnectionParams", () => {
+  let client: any;
+  const accessToken = "dummy-access-token-1234";
+
+  before(() => {
+    client = createClient();
+    client.auth.tokenManager.setToken(accessToken);
+  });
+
+  it("should return an object with proper Authorization property", () => {
+    expect(client.graphQL.getConnectionParams()).to.deep.equal({
+      Authorization: `Bearer ${accessToken}`
+    });
+  });
+
+  describe("when access token is missing", () => {
+    before(() => {
+      client.auth.tokenManager._token = undefined;
+    })
+
+    it("should still return proper connection params", () => {
+      expect(client.graphQL.getConnectionParams()).to.deep.equal({
+        Authorization: "Bearer undefined"
```

Comment:
This test is very silly, this can't happen in practice but I needed to add this case for 100% coverage on the feature 🤷‍♂

Reply:
Or let's skip that helper method at all (see my inline comment).